OneSignalSDK/unittest/src/test/java/com/test/onesignal/SynchronizerIntegrationTests.java

```diff
@@ -0,0 +1,1831 @@
+    private static OneSignal.EmailUpdateError lastEmailUpdateFailure;
+    private static OneSignal.EmailUpdateHandler getEmailUpdateHandler() {
+        return new OneSignal.EmailUpdateHandler() {
+            @Override
+            public void onSuccess() {
+                didEmailUpdateSucceed = true;
+            }
+
+            @Override
+            public void onFailure(OneSignal.EmailUpdateError error) {
+                lastEmailUpdateFailure = error;
+            }
+        };
+    }
+
+    private static JSONObject didSMSlUpdateSucceed;
+    private static OneSignal.OSSMSUpdateError lastSMSUpdateFailure;
+    private static OneSignal.OSSMSUpdateHandler getSMSUpdateHandler() {
+        return new OneSignal.OSSMSUpdateHandler() {
+            @Override
+            public void onSuccess(JSONObject result) {
+                didSMSlUpdateSucceed = result;
+            }
+
+            @Override
+            public void onFailure(OneSignal.OSSMSUpdateError error) {
+                lastSMSUpdateFailure = error;
+            }
+        };
+    }
+
+    private static void cleanUp() throws Exception {
+        lastExternalUserIdResponse = null;
+        lastExternalUserIdError = null;
+        lastEmailUpdateFailure = null;
+        didEmailUpdateSucceed = false;
+
+        ShadowGMSLocationController.reset();
+
+        TestHelpers.beforeTestInitAndCleanup();
+
+        // Set remote_params GET response
+        setRemoteParamsGetHtmlResponse();
+    }
+
+    @BeforeClass // Runs only once, before any tests
+    public static void setUpClass() throws Exception {
+        ShadowLog.stream = System.out;
+
+        TestHelpers.beforeTestSuite();
+
+        Field OneSignal_CurrentSubscription = OneSignal.class.getDeclaredField("subscribableStatus");
+        OneSignal_CurrentSubscription.setAccessible(true);
+
+        OneSignal.setLogLevel(OneSignal.LOG_LEVEL.VERBOSE, OneSignal.LOG_LEVEL.NONE);
+        StaticResetHelper.saveStaticValues();
+    }
+
+    @Before
+    public void beforeEachTest() throws Exception {
+        blankActivityController = Robolectric.buildActivity(BlankActivity.class).create();
+        blankActivity = blankActivityController.get();
+        time = new MockOSTimeImpl();
+        trackerFactory = new OSTrackerFactory(new MockOSSharedPreferences(), new MockOSLog(), time);
+        sessionManager = new MockSessionManager(OneSignal_getSessionListener(), trackerFactory, new MockOSLog());
+        dbHelper = new MockOneSignalDBHelper(ApplicationProvider.getApplicationContext());
+
+        TestHelpers.setupTestWorkManager(blankActivity);
+
+        cleanUp();
+
+        OneSignal_setTime(time);
+    }
+
+    @After
+    public void afterEachTest() throws Exception {
+        afterTestCleanup();
+    }
+
+    @AfterClass
+    public static void afterEverything() throws Exception {
+        cleanUp();
+    }
+
+    @Test
+    public void shouldSetEmail() throws Exception {
+        OneSignalInit();
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.